langchain/callbacks/context_callback.py

```diff
@@ -0,0 +1,193 @@
+    except ImportError:
+        raise ImportError(
+            "To use the context callback manager you need to have the "
+            "`getcontext` python package installed (version >=0.3.0). "
+            "Please install it with `pip install --upgrade python-context`"
+        )
+    return getcontext, Credential, Conversation, Message, MessageRole, Rating
+
+
+class ContextCallbackHandler(BaseCallbackHandler):
+    """Callback Handler that records transcripts to Context (https://getcontext.ai).
+
+    Args:
+        token: The token with which to authenticate requests to Context.
+            Visit https://go.getcontext.ai/settings to generate a token.
+
+    Raises:
+        ImportError: if the `context-python` package is not installed.
+
+    Chat Example:
+        >>> from langchain.llms import ChatOpenAI
+        >>> from langchain.callbacks import ContextCallbackHandler
+        >>> context_callback = ContextCallbackHandler(
+        ...     token="<CONTEXT_TOKEN_HERE>",
+        ... )
+        >>> chat = ChatOpenAI(
+        ...     temperature=0,
+        ...     headers={"user_id": "123"},
+        ...     callbacks=[context_callback],
+        ...     openai_api_key="API_KEY_HERE",
+        ... )
+        >>> messages = [
+        ...     SystemMessage(content="You translate English to French."),
+        ...     HumanMessage(content="I love programming with LangChain."),
+        ... ]
+        >>> chat(messages)
+
+    Chain Example:
+        >>> from langchain import LLMChain
+        >>> from langchain.llms import ChatOpenAI
+        >>> from langchain.callbacks import ContextCallbackHandler
+        >>> context_callback = ContextCallbackHandler(
+        ...     token="<CONTEXT_TOKEN_HERE>",
+        ... )
+        >>> human_message_prompt = HumanMessagePromptTemplate(
+        ...     prompt=PromptTemplate(
+        ...         template="What is a good name for a company that makes {product}?",
+        ...         input_variables=["product"],
+        ...    ),
+        ... )
+        >>> chat_prompt_template = ChatPromptTemplate.from_messages(
+        ...   [human_message_prompt]
+        ... )
+        >>> callback = ContextCallbackHandler(token)
+        >>> # Note: the same callback object must be shared between the
+        ...   LLM and the chain.
+        >>> chat = ChatOpenAI(temperature=0.9, callbacks=[callback])
+        >>> chain = LLMChain(
+        ...   llm=chat,
+        ...   prompt=chat_prompt_template,
+        ...   callbacks=[callback]
+        ... )
+        >>> chain.run("colorful socks")
+    """
+
+    def __init__(self, token: str, verbose: bool = False, **kwargs: Any) -> None:
+        (
+            self.context,
+            self.credential,
+            self.conversation_model,
+            self.message_model,
+            self.message_role_model,
+            self.rating_model,
+        ) = import_context()
+
+        self.client = self.context.ContextAPI(credential=self.credential(token))
```

Comment:
up to you if you want to do this, in other places we usually provide an option to specify api tokens via env vars. could do something like
```python
import os

def __init__(self, token: Optional[str] = None,...):
  ...
  token = token or os.environ.get("CONTEXT_API_TOKEN")
  ...

Reply:
Nice idea, have added.